server/sonar-server/src/main/java/org/sonar/server/measure/index/ProjectMeasuresIndex.java

```diff
@@ -304,6 +312,34 @@ private static QueryBuilder toValueQuery(MetricCriterion criterion) {
     }
   }
 
+  public List<String> searchTags(@Nullable String textQuery, int pageSize) {
```

Comment:
Why only pageSize ? Either you should have page and pageSize, either you should return everything